core/src/main/java/io/elsci/assertreflectionequals/BuildErrorMessage.java

```diff
@@ -1,34 +1,180 @@
 package io.elsci.assertreflectionequals;
 
-import java.util.Deque;
-import java.util.Iterator;
+import java.lang.reflect.Field;
+import java.util.*;
 
 class BuildErrorMessage {
-    public static StringBuilder build(Deque<Class<?>> fullPath, Object expectedValue, String expectedName,
-                                   Object actualValue, StringBuilder errorMessage) {
-        errorMessage.append("Values were different for: ");
-        Iterator<Class<?>> iterator = fullPath.descendingIterator();
-        while (iterator.hasNext()) {
-            errorMessage.append(iterator.next().getSimpleName()).append(".");
+
+    /**
+     * Builds error for cases when one of the comparing objects is null or initial objects don't belong to the same class. Example:
+     * @param initialObjects initial (expected and actual) objects
+     * @param expectedObject initial or child expected object
+     * @param actualObject initial or child actual object
+     */
+    public static void build(List<Object> initialObjects, Deque<String> fullPath, Object expectedObject, Object actualObject, StringBuilder errorMessage) {
```

Comment:
There are so many of the build() methods - all of them are slightly different. It's hard to keep track which one is for what. Maybe it's possible to simplify things? For instance:

1. Maybe instead of static methods, we can create an ErrorMessageBuilder and keep initialObjects, fullPath and errorMessage as fields?
2. And what if we gave names to the build() methods? Like `builder.addInitialPart().addDifferentArrayValues(...)`?

Reply:
Yes, of course I was thinking about renaming methods as a part of improvements, but not about the 1st option.
Applied both ideas, but this is still not polished and w/o extra unit tests.